### DISCUSSION TITLE
ESP32 + Doorbell on GPIOs. Permament Ghosting issue

### DISCUSSION THREAD

**Discussion by qrioniclabs** [arendst/Tasmota#17065] [Q&A] [answered]
Hello,

I need some help to get my doorbells connected to Tasmota properly. Tasmota is running on an ESP32-Wroom-32 by AZ Delivery (https://www.az-delivery.de/en/products/esp32-developmentboard). The doorbells are "MISTRAL SE 03 Funksender" (by German company Grothe). Connected to each doorbell's inputs is a dry contact switch which is mounted at our gate. The actual doorbells are needed because they relay their signal to a remotely connected battery powered bell by the same company (used by the parents).

So I connected GND and GPIO 18 / 19 in parallel to those contacts. With my multimeter I am reading a constant current of 2.4V coming from each of the doorbells. When the doorbell's button is pressed, I read 0.0V.

At first it looked like it was working. When I pressed the button on the doorbell, Tasmota switched the appropriate relay and also the remote bell chimes.

But after I while I am getting Ghosting issues. For the doorbell at GPIO18, Tasmota recognizes toggles permanently:

```
11:57:43.637 MQT: stat/w-thing-doorbell/RESULT = {"POWER1":"ON"}
11:57:43.641 MQT: stat/w-thing-doorbell/POWER1 = ON
11:57:58.227 MQT: stat/w-thing-doorbell/RESULT = {"POWER1":"OFF"}
11:57:58.232 MQT: stat/w-thing-doorbell/POWER1 = OFF
11:57:58.376 MQT: stat/w-thing-doorbell/RESULT = {"POWER1":"ON"}
11:57:58.382 MQT: stat/w-thing-doorbell/POWER1 = ON
11:58:01.627 MQT: stat/w-thing-doorbell/RESULT = {"POWER1":"OFF"}
11:58:01.632 MQT: stat/w-thing-doorbell/POWER1 = OFF
11:58:21.622 MQT: stat/w-thing-doorbell/RESULT = {"POWER1":"ON"}
11:58:21.628 MQT: stat/w-thing-doorbell/POWER1 = ON
11:58:24.732 MQT: stat/w-thing-doorbell/RESULT = {"POWER1":"OFF"}
11:58:24.738 MQT: stat/w-thing-doorbell/POWER1 = OFF
11:58:25.703 MQT: stat/w-thing-doorbell/RESULT = {"POWER1":"ON"}
11:58:25.708 MQT: stat/w-thing-doorbell/POWER1 = ON
11:58:33.584 MQT: stat/w-thing-doorbell/RESULT = {"POWER1":"OFF"}
11:58:33.593 MQT: stat/w-thing-doorbell/POWER1 = OFF
```
For the doorbell connected to GPIO19 it's much less of an issue, but with that I get a ghosted presses too, maybe a couple of times a day.

Any ideas what could be the problem here? Is it technically correct to expect Tasmota recognizing the 2.4V input or is it expecting 5V and so recognizes 2.4 as LOW? Do I need additional electrical components or can I configure those ports differently? Or use another port?

The pinout for the ESP32 can be found here: https://cdn.shopify.com/s/files/1/1509/1638/files/ESP-32_NodeMCU_Developmentboard_Pinout.pdf?v=1609851295

Using Tasmota 12.2.0 and MQTT.

Thanks for you help!

**Comment by barbudor** [comment, marked as answer]
You don't tell anything about how you have configured Tasmota (buttons, switches, relays....)
And a little wiring diagram is generally a good start to explain what you've done

Now connecting the ESP32 in parrallel of an existing circuit is calling stringly for magic smoke. You may have damaged your ESP32 irremediably.

What you should have done is replicate the dry contact scheme by using the push button at the gate to go ONLY to Tasmota and having Tasmota to drive a relay (or an optocoupler) to action the existing "MISTRAL SE 03" in place of the gate button. This would have insured perfect isolation between the circuits.

**Reply by qrioniclabs** [reply]
Thanks for your answer. My thinking was just dumb. I will go back to the drawing board and set it up the way you wrote.
